src/test/java/com/sap/cloudfoundry/client/facade/ApplicationsCloudControllerClientIntegrationTest.java

```diff
@@ -0,0 +1,441 @@
+    void testApplicationEnvironment() {
+        String applicationName = "test-application-4";
+        try {
+            verifyApplicationWillBeCreated(applicationName, ImmutableStaging.builder()
+                                                                            .build(),
+                                           Set.of(ImmutableCloudRouteSummary.builder()
+                                                                            .host(APPLICATION_HOST)
+                                                                            .domain(DEFAULT_DOMAIN)
+                                                                            .build()));
+            client.updateApplicationEnv(applicationName, Map.of("foo", "bar"));
+            UUID applicationGuid = client.getApplicationGuid(applicationName);
+            Map<String, String> applicationEnvironment = client.getApplicationEnvironment(applicationGuid);
+            assertEquals("bar", applicationEnvironment.get("foo"));
+        } finally {
+            client.deleteApplication(applicationName);
+        }
+    }
+
+    @Test
+    @DisplayName("Create application, upload a package, verify package exists")
+    void uploadApplication() throws URISyntaxException, InterruptedException {
+        String applicationName = "test-application-5";
+        try {
+            verifyApplicationWillBeCreated(applicationName, ImmutableStaging.builder()
+                                                                            .build(),
+                                           Set.of(ImmutableCloudRouteSummary.builder()
+                                                                            .host(APPLICATION_HOST)
+                                                                            .domain(DEFAULT_DOMAIN)
+                                                                            .build()));
+            CloudPackage cloudPackage = uploadApplication(applicationName);
+            assertEquals(Status.READY, cloudPackage.getStatus());
+        } finally {
+            client.deleteApplication(applicationName);
+        }
+    }
+
+    @Test
+    @DisplayName("Create application, upload a package, create a build and restart the application")
+    void restartApplication() throws URISyntaxException, InterruptedException {
+        String applicationName = "test-application-6";
+        try {
+            verifyApplicationWillBeCreated(applicationName, ImmutableStaging.builder()
+                                                                            .build(),
+                                           Set.of(ImmutableCloudRouteSummary.builder()
+                                                                            .host(APPLICATION_HOST)
+                                                                            .domain(DEFAULT_DOMAIN)
+                                                                            .build()));
```

Comment:
This is used in almost every test. What about extracting it in a method? `createAndVerifyDefaultApplication(String appName)`?